STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+# Front-end style guide
+
+This document outlines recommended code style, quirks, and best practices for front-end development on Kedro-Viz.
```

Comment:
I'd maybe open with something that explains who it's for. Sort of like this perhaps?

```suggestion
Here are our recommendations for anyone contributing front-end development to Kedro-Viz; we cover code style, quirks, and best practices.
```